pkg/controller/memberstatus/memberstatus_controller_test.go

```diff
@@ -0,0 +1,284 @@
+package memberstatus
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/codeready-toolchain/api/pkg/apis"
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	. "github.com/codeready-toolchain/member-operator/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var requeueResult = reconcile.Result{RequeueAfter: defaultRequeueTime}
+
+const defaultMemberOperatorName = "member-operator"
+
+func TestNoMemberStatusFound(t *testing.T) {
+	t.Run("No memberstatus resource found", func(t *testing.T) {
+		// given
+		requestName := "bad-name"
+		getHostClusterFunc := newGetHostClusterReady
+		reconciler, req, _ := prepareReconcile(t, requestName, getHostClusterFunc)
+
+		// when
+		res, err := reconciler.Reconcile(req)
+
+		// then - there should not be any error, the controller should only log that the resource was not found
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{}, res)
+	})
+
+	t.Run("No memberstatus resource found - right name but not found", func(t *testing.T) {
+		// given
+		expectedErrMsg := "get failed"
+		requestName := defaultMemberStatusName
+		getHostClusterFunc := newGetHostClusterReady
+		reconciler, req, fakeClient := prepareReconcile(t, requestName, getHostClusterFunc)
+		fakeClient.MockGet = func(ctx context.Context, key types.NamespacedName, obj runtime.Object) error {
+			return fmt.Errorf(expectedErrMsg)
+		}
+
+		// when
+		res, err := reconciler.Reconcile(req)
+
+		// then - there should not be any error, the controller should only log that the resource was not found
+		require.Error(t, err)
+		require.Equal(t, expectedErrMsg, err.Error())
+		assert.Equal(t, reconcile.Result{}, res)
+	})
```

Comment:
Also check the status.

Reply:
In this case I am not checking the status because there is no `memberStatus` resource. Unless you mean to check the status of something else?

Reply:
Never mind. I confused it with the No Deployment test.